ext/pdo_firebird/firebird_driver.c

```diff
@@ -848,7 +904,8 @@ static bool firebird_handle_set_attribute(pdo_dbh_t *dbh, zend_long attr, zval *
 						if (bval) {
 							/* turning on auto_commit with an open transaction is illegal, because
 							   we won't know what to do with it */
-							H->last_app_error = "Cannot enable auto-commit while a transaction is already open";
+							const char *msg = "Cannot enable auto-commit while a transaction is already open";
+							firebird_error_with_info(dbh, "HY000", strlen("HY000"), msg, strlen(msg));
```

Comment:
@nielsdos just to sanity check, but this is sensible right? The compiler will replace at compile time the strlen call by the actual value

Reply:
Yes it does

Reply:
Thank you both! I was able to understand it a little. I'll learn some more.

Reply:
I tried various things. I understand that `strlen("HY000")` will be replaced, but `strlen(msg)` will not be replaced, is this correct?

Reply:
Yes, because ``msg`` is a variable, where the content is only known at runtime.

Reply:
I didn't look at this specific case, but in this one the compiler should be smart enough :)